File: web-sources/S8-fwks-palm/boxes/paper/PaperColumn.js



import { forSubComponents } from "/S8-pkgs-ui-carbide/carbide.js";
import { S8 } from "/S8-api/S8Context.js";
import { S8Object } from "/S8-api/S8Object.js";

import { S8WebFront } from "/S8-pkgs-ui-carbide/S8WebFront.js";

S8.page.CSS_import('/S8-fwks-palm/boxes/paper/PaperStruct.css');


/**
 * 
 */
export class PaperColumn extends S8Object {

    /** @type {HTMLDivElement} */
    wrapperNode;

    /**
     * 
     */
    constructor() {
        super();
        this.wrapperNode = document.createElement("div");
        this.wrapperNode.classList.add("paperstruct-column");

        /*
        this.extensionNode = document.createElement("div");
        this.wrapperNode.appendChild(this.extensionNode);
        */        

        const _this = this;
        this.wrapperNode.addEventListener("scroll", function(event){  
            forSubComponents(_this, function(object){
                if(object.onScroll){ object.onScroll(event); }
            });
        }, false);
    }


    /**
     * 
     * @returns 
     */
    getEnvelope() {
        return this.wrapperNode;
    }



    /** @returns{S8Object[]} */
    getComponents() {
        return this.elements;
    }


    /** @param {S8Object[]} elements */
    S8_set_elements(elements) {
        this.elements = elements;
        S8WebFront.setS8Elements(this.wrapperNode, elements);
    }


    /**
     * 
     * @param {number} width 
     */
    S8_set_width(width) {
        if (width < 0) { /* clear width */
            this.wrapperNode.style.removeProperty("width");
        }
        else if (width > 0) {
            this.wrapperNode.style.width = `${width * 100}vw`;
        }
    }


    /** render */
    S8_render() { }


    /** dispose */
    S8_dispose() { }

}

File: sources/com/s8/fwks/palm/boxes/paper/PaperColumn.java
package com.s8.fwks.palm.boxes.paper;

import com.s8.api.web.S8WebFront;
import com.s8.api.web.S8WebObject;
import com.s8.fwks.palm.PalmWebSources;


/**
 * 
 * @author pierreconvert
 *
 */
public class PaperColumn extends S8WebObject {

	
	/**
	 * 
	 * @param branch
	 */
	public PaperColumn(S8WebFront branch) {
		super(branch, PalmWebSources.ROOT_WEBPATH + "/boxes/paper/PaperColumn");
	}
	
	
	
	/**
	 * 
	 * @param name
	 */
	public void setElements(S8WebObject... elements) {
		vertex.outbound().setObjectListField("elements", elements);
	}
	
	
	public void setWidth(double width) {
		vertex.outbound().setFloat32Field("width", (float) width);
	}
	
}
